src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -6736,6 +6739,12 @@ int RGWRados::Object::Read::prepare(optional_yield y, const DoutPrefixProvider *
   if (r < 0)
     return r;
 
+  if (manifest /* params.parts_count */) {
+      RGWObjManifest::obj_iterator end = manifest->obj_end(dpp);
+      auto cur_part_id = end.get_cur_part_id();
+      params.parts_count = (cur_part_id == 1) ? 1 : cur_part_id - 1;;
```

Comment:
the comment now reads `If the object is multipart, the total number of multipart parts is assigned to this output parameter.`, but it looks like this sets a `parts_count` for non-multipart uploads too. can we find a way to leave `parts_count` as an empty optional when it's not a multipart manifest?

Reply:
is that really helpful?

Reply:
maybe it's easy

Reply:
the general reason that this logic would have a manifest, is that this is a multipart upload.  I don't think this really needs to be changed.

Reply:
i'm concerned that this is going to cause Get/HeadObject requests to start returning x-amz-mp-parts-count headers for non-multipart uploads where they didn't before, and that this doesn't do what the comment says it does

Reply:
it appears that RWObjManifest has a RGWObjTier which has a boolean member "is_multipart_upload".  I do not know if it would be safe or correct to add that to the conditional.

Reply:
well the comment can be changed.  but afaik, the only potential issue would be with non-multipart objects with a manifest.  do I have a way to test for that?  doing so should resolve that issue

Reply:
`end.get_cur_part_id()` returns 0 for non-multipart manifests

Reply:
ok

Reply:
done